File: src/Utt2Database/PlayerLog.vb
Imports System
Imports System.Collections.Generic
Imports Microsoft.VisualBasic

Namespace Utt2Database
    Partial Public Class PlayerLog
        Public Property Id As Long?

        Public Property PlayerId As Long

        Public Property ServerId As Long

        Public Property MatchId As Long

        Public Property SeenCount As Integer

        Public Property LastSeenTime As Date

        Public Property FirstSeenTime As Date

        Public Property ScoreThisMatch As Long

        Public Property DeathsThisMatch As Integer?

        Public Property PingSum As Integer

        Public Property Team As Integer

        Public Property Finished As Boolean

        Public Overridable Property Match As ServerMatch

        Public Overridable Property Player As Player

        Public Overridable Property Server As Server

        Public Overrides Function ToString() As String
            Return Player.Name & "#M" & MatchId & "@S" & ServerId
        End Function
    End Class
End Namespace

File: src/IndexerSharp/Utt2Database/PlayerLog.cs
using Microsoft.EntityFrameworkCore.Metadata.Internal;

namespace Naomai.UTT.Indexer.Utt2Database;
public partial class PlayerLog
{
    public long? Id { get; set; }
    public long PlayerId { get; set; }
    public long ServerId { get; set; }
    public long MatchId { get; set; }
    public int SeenCount { get; set; }
    public DateTime LastSeenTime { get; set; }
    public DateTime FirstSeenTime { get; set; }
    public long ScoreThisMatch { get; set; }
    public int? DeathsThisMatch { get; set; }
    public int PingSum { get; set; }
    public int Team { get; set; }
    public bool Finished { get; set; }
    public ServerMatch Match { get; set; } = null!;
    public Player Player { get; set; } = null!;
    public Server Server { get; set; } = null!;

    public override string ToString()
    {
        return "PL" +PlayerId + "#M" + MatchId + "@S" + ServerId;
    }
}